graphql/resolver.go

```diff
@@ -38,14 +38,28 @@ func New(apiURL string) Config {
 			return next(ctx)
 		}
 
-		// Check if this call is for create or copy project
+		// Check for admin permissions for each of the resolvers.
 		args, isStringMap := obj.(map[string]interface{})
 		if !isStringMap {
 			return nil, ResourceNotFound.Send(ctx, "Project not specified")
 		}
-		projectIdToCopy, hasProjectId := args["project"].(map[string]interface{})["projectIdToCopy"].(string)
-		if hasProjectId {
-			// Check if the user has permission to copy the project
+		operationContext := graphql.GetOperationContext(ctx).OperationName
+
+		if operationContext == "CreateProject" {
```

Comment:
Yes, it's directly related. What the directive was doing before was checking if the function arg `projectIdToCopy` existed, which only belongs to the `copyProject` mutation; so if it didn't exist it would mean the `createProject` mutation was being called

Reply:
can we make these constants? Maybe top of file?

Reply:
(alternatively, should we have more directives than just one? Could be a different ticket though)